### DISCUSSION TITLE
RunHarmony() warning - Invalid name supplied, making object name syntactically valid

### DISCUSSION THREAD

**Discussion by lucygarner** [satijalab/seurat#6565] [Q&A]
Hi,

I am running Harmony as follows:
```
seurat_object <- RunHarmony(
    seurat_object, 
    group.by.vars = "donor", 
    dims.use = 1:40, 
    plot_convergence = TRUE, 
    assay.use = "SCT", 
    theta = theta_value, 
    reduction.save = "harmony_40pc"),
    verbose = TRUE)
```

I am getting the following warning, but I can't see why the name of my reduction is not syntactically valid. The examples for reduction names are similar to mine e.g. 'harmony_theta2'.

> Warning: Invalid name supplied, making object name syntactically valid. New object name is Seurat..ProjectDim.SCT.harmony_40pc; see ?make.names for more details on syntax validity

Any thoughts?

Best wishes,
Lucy

**Comment by ShanglaiLi** [comment]
hi 
l had the same problem,
did you solve it?
the error:Error: (converted from warning) Invalid name supplied, making object name syntactically valid. New object name is Seurat.. ProjectDim.RNA.harmony; see ?make.names for more details on syntax validity

**Reply by ShanglaiLi** [reply]
hi，
I solve it，you need add "project.dim = F",the warning was disappear.
```scRNA_harmony <- RunHarmony (scRNA_harmony, group.by.vars = "orig.ident",project.dim = F)```

**Reply by lucygarner** [reply]
That changes the way the function works though, so I don't want to do that. It's just a warning, so I just carry on.

**Reply by TIPPLERXIE** [reply]
so are there any problems when you ignore it?
